tests/Application/Actions/LoginTest.php

```diff
@@ -36,6 +36,9 @@ public function testSuccess(): void
             ->shouldBeCalledOnce()
             ->willReturn($personWithHashMatchingRawPasswordFromLoginObject);
 
+        $personRepoProphecy->upgradePasswordIfPossible($person->raw_password, Argument::type(Person::class))
```

Comment:
We can be specific about which person it should be called for:
```suggestion
        $personRepoProphecy->upgradePasswordIfPossible($person->raw_password, $personWithHashMatchingRawPasswordFromLoginObject)
```

Reply:
Thanks! I didn't read the existing test properly and had tried the wrong var.